### PR TITLE
Added How To use FastaiLRFinder Guide

### LINKED ISSUE #5: Create new how-to guide for FastAI LRFinder
Shift the notebook [FastaiLRFinder_MNIST.ipynb](https://github.com/pytorch/ignite/blob/master/examples/notebooks/FastaiLRFinder_MNIST.ipynb) as a how-to guide making appropriate changes.

### PR DESCRIPTION
**Pull request by Priyansi**
Fixes #5 
Moved [FastaiLRFinder_MNIST.ipynb](https://github.com/pytorch/ignite/blob/master/examples/notebooks/FastaiLRFinder_MNIST.ipynb) notebook to how-to guides.
Changes in the flow of the notebook, explanation, comments and code were made. ProgressBar handler was removed to avoid unnecessary code.

### CODE REVIEW

**Comment by sdesrozis** [comment]
@Priyansi it looks good to me !

File: how-to-guides/fastai-lr-finder.ipynb [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+   ],
+   "outputs": [],
+   "metadata": {
+    "id": "CTGJPVI6mvd2"
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "When attached to the `trainer`, this handler follows the same procedure used by [fastai](https://docs.fast.ai/callback.schedule.html#LRFinder). The model is trained for `num_iter` iterations while the learning rate is increased from its initial value specified by the optimizer to the upper bound called `end_lr`. This increase can be linear (`step_mode=\"linear\"`) or exponential (`step_mode=\"exp\"`). The default `step_mode` is exponential which is recommended for larger learning rate ranges while linear provides good results for small ranges."
```
Comment (KickItLikeShika) [inline how-to-guides/fastai-lr-finder.ipynb:224]: > learning rate is increased from its initial value specified by the optimizer to the upper bound called `end_lr`

That's not 100% true anymore, we recently added the start_lr option, see here https://github.com/pytorch/ignite/pull/2111
, so I think saying mentioning it will increase from `start_lr` to `end_lr` and if `start_lr` is None it will start from LR with optimizer be default would be better.

**Review by KickItLikeShika — COMMENTED** [review]
Thanks a lot for the PR @Priyansi, very nice work! I just have one comment.

File: how-to-guides/fastai-lr-finder.ipynb (hunk 1 above)
Comment (Priyansi) [reply how-to-guides/fastai-lr-finder.ipynb:224]: Thank you for pointing that out! I'll make the change.

**Review by trsvchn — COMMENTED** [review]
Thanks @Priyansi !  Good work! I left some comments

> We will compare the results produced with and without using the handler for better understanding.

I think I didn't the compare results part. Because there is no model reinitialization after the first run with the constant lr, so I see that you run the same model on 6 epochs: 3 w/ constant lr + 3 with proposed lr.

**Comment by trsvchn** [comment]
Maybe we can Add more headers: E.g: `## Constant LR`, `LR Finder` or alternatively remove `## Basic Setup`, because we have just one related to setup and that's weird

File: how-to-guides/fastai-lr-finder.ipynb [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+    "\n",
+    "test_loader = DataLoader(\n",
+    "    MNIST(download=True, root=\".\", transform=data_transform, train=False),\n",
+    "    batch_size=256,\n",
+    "    shuffle=False,\n",
+    ")\n",
+    "\n",
+    "\n",
+    "model = Net().to(device)\n",
+    "optimizer = torch.optim.RMSprop(model.parameters(), lr=1e-06)\n",
```
[the remark this answers is not in the document]
Comment (Priyansi) [reply how-to-guides/fastai-lr-finder.ipynb:103]: Because I couldn't show a good enough improvement, only about 0.5% increase in accuracy. Mainly because it's hard to mess up MNIST. So I set the fixed lr very low.

**Comment by Priyansi** [comment]
Thank you for the comments @trsvchn ! I updated the notebook.

File: how-to-guides/fastai-lr-finder.ipynb [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+  {
+   "cell_type": "markdown",
+   "source": [
+    "<!-- ---\n",
+    "title: How to use FastAILRFinder with Ignite\n",
+    "downloads: true\n",
+    "sidebar: true\n",
+    "tags:\n",
+    "  - lr finder\n",
+    "  - fastai lr finder\n",
```
Comment (vfdev-5) [inline how-to-guides/fastai-lr-finder.ipynb:12]:
```suggestion
```

File: how-to-guides/fastai-lr-finder.ipynb [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+    "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
+    "\n",
+    "\n",
+    "class Net(nn.Module):\n",
+    "    def __init__(self):\n",
+    "        super(Net, self).__init__()\n",
+    "\n",
+    "        self.model = resnet18(num_classes=10)\n",
+    "        self.model.conv1 = nn.Conv2d(\n",
+    "            1, 64, kernel_size=(3, 3), stride=(2, 2), padding=(3, 3), bias=False\n",
```
Comment (vfdev-5) [inline how-to-guides/fastai-lr-finder.ipynb:76]: Why do we need to pad as 3 if kernel size is 3, should be padding 1

File: how-to-guides/fastai-lr-finder.ipynb [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "mophHZUkICKI",
+    "outputId": "378c7aac-b3c8-49ff-8be1-acc547730c12"
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Let's see how we can achieve better results by using the `FastAILRFinder` handler. But first, let's restore the initial state of the model and optimizer so we can re-train them from scratch. "
```
Comment (vfdev-5) [inline how-to-guides/fastai-lr-finder.ipynb:208]:
```suggestion
    "Let's see how we can achieve better results by using the `FastaiLRFinder` handler. But first, let's restore the initial state of the model and optimizer so we can re-train them from scratch. "
```

And let's add a link to our docs of FastaiLRFinder

**Comment by Priyansi** [comment]
Thank you @vfdev-5 for the review, I have made the suggested changes.

**Review by vfdev-5 — APPROVED** [review]
LGTM
